agent/cmd/cuda-checkpoint-helper/daemon_protocol_test.cpp

```
@@ -0,0 +1,536 @@
+}
+
+void TestGoldenRequestFixture() {
+  const std::vector<unsigned char> encoded = ReadGoldenRequest();
+  Request parsed;
+  std::string error;
+  assert(ParseRequest(encoded.data(), encoded.size(), &parsed, &error));
+  assert(parsed.action == Action::kRestore);
+  assert(parsed.backend == Backend::kPosix);
+  assert(parsed.pid == 42U);
+  assert(parsed.transfer_buffer_count == 2U);
+  assert(parsed.transfer_chunk_bytes == 8U * 1024U * 1024U);
```

Clang (14.0.6)

[warning] performing an implicit widening conversion to type 'unsigned long' of a multiplication performed in type 'unsigned int'

(bugprone-implicit-widening-of-multiplication-result)